modules/Bio/EnsEMBL/Variation/DBSQL/LDFeatureContainerAdaptor.pm

```diff
@@ -454,6 +435,8 @@ sub _fetch_by_Slice_VCF {
           $sample_count
         ) = split /\s/;
 
+        # filter by r2 and d_prime values
+        next if ($r2 < $self->r2 || $d_prime < $self->d_prime);
```

Comment:
You could put $self->r2 and $self->d_prime in variables outside the loop to avoid calling the methods every loop.